lib/platform/linux/logger.c

```diff
@@ -8,21 +8,61 @@
 
 #include <time.h>
 
-static inline void get_timestamp(char timestamp[26])
+/* Full log level string  */
+static char DEBUG[] = "DEBUG";
+static char INFO[] = "INFO";
+static char WARNING[] = "WARNING";
+static char ERROR[] = "ERROR";
+
+static inline void get_timestamp(char timestamp[37])
 {
-    time_t ltime;
     struct tm result;
+    long ms;
+    time_t s;
+    struct timespec spec;
+
+    clock_gettime(CLOCK_REALTIME, &spec);
+
+    s = spec.tv_sec;
+    localtime_r(&s, &result);
+
+    ms = spec.tv_nsec / 1.0e6;  // Convert nanoseconds to milliseconds
 
-    ltime = time(NULL);
-    localtime_r(&ltime, &result);
-    sprintf(timestamp, "%d:%d:%d", result.tm_hour, result.tm_min, result.tm_sec);
+    sprintf(timestamp,
+            "%d-%d-%d %d:%d:%d.%03ld",
```

Comment:
Not that it makes a big difference, but the transport service is presenting a comma between the seconds and ms

[see log at](https://travis-ci.com/wirepas/gateway):
wm-transport    | 2019-10-30 22:03:07,820 | [DEBUG] wirepas_gateway@mqtt_wrapper.py:225:Subscribing to: gw-request/send_data/2485377957891/+

Reply:
Yes! TBH, I saw it but forget to modify it ;-) I'll do